hedera-node/src/main/java/com/hedera/services/store/tokens/unique/UniqueTokenStore.java

```diff
@@ -63,34 +71,69 @@ public UniqueTokenStore(final EntityIdSource ids,
 	) {
 		super(ids, validator, properties, tokens, tokenRelsLedger);
 		this.uniqueTokenSupplier = uniqueTokenSupplier;
+		this.provisionalUniqueTokens = new ArrayList<>(); // TODO MAX_SIZE
+		this.lastMintedSerialNumbers = new ArrayList<>();
 	}
 
 	@Override
-	public ResponseCodeEnum mint(final TokenID tId, final String memo, final RichInstant creationTime) {
-		return tokenSanityCheck(tId, (merkleToken -> {
+	public List<Long> popLastMintedSerialNumbers() {
+		var copy = new ArrayList<>(lastMintedSerialNumbers);
+		lastMintedSerialNumbers.clear();
+		return copy;
+	}
+
+	@Override
+	public ResponseCodeEnum wipe(final AccountID aId, final TokenID tId, final long wipingAmount, final boolean skipKeyCheck) {
+		return null;
+	}
+
+	@Override
+	public ResponseCodeEnum mint(final TokenMintTransactionBody txBody, final RichInstant creationTime) {
+		return tokenSanityCheck(txBody.getToken(), merkleToken -> {
```

Comment:
We are using `txBody.getToken()` a couple of times. Maybe we can extract it to a variable `tokenId`?